pkg/cloud-provider/cloudapi/aws/aws_converters.go

```diff
@@ -134,12 +157,20 @@ func convertFromIPPermissionToIngressRule(ipPermissions []*ec2.IpPermission, man
 }
 
 // convertFromIPPermissionToEgressRule converts cloud egress rules from ec2.IpPermission to internal securitygroup.EgressRule.
+// Each AT Sg can have one or more ANPs and an ANP can have one or more rules. Each rule can have a description.
 func convertFromIPPermissionToEgressRule(ipPermissions []*ec2.IpPermission, managedSGs map[string]*ec2.SecurityGroup,
 	unmanagedSGs map[string]*ec2.SecurityGroup) []securitygroup.EgressRule {
 	var egressRules []securitygroup.EgressRule
 	for _, ipPermission := range ipPermissions {
-		toDstIPs := convertFromIPRange(ipPermission.IpRanges)
-		for _, dstIP := range toDstIPs {
+		toDstIPs, desc := convertFromIPRange(ipPermission.IpRanges)
+		for i, dstIP := range toDstIPs {
+			// Get cloud rule description.
+			_, ok := securitygroup.ExtractCloudDescription(desc[i])
+			if !ok {
+				// Ignore rules that don't have a valid description field.
+				awsPluginLogger().V(4).Info("Failed to extract cloud rule description", "desc", desc[i])
+				continue
```

Comment:
I believe this is used in cloudSync, so in this empty description case we should still converts the cloud user created rule, and report it to the cloudRule indexer. Otherwise the updateANPRules won't account for this user created rule and might lead to duplicate rules not caught.

Reply:
This is an initial change to handle description field in cloud.
With this change intension is to find a way to distinguish user created rule vs nephe created rule.

Now coming to logical change, which will be done in a subsequent patch.
how do you envision handling user created vs nephe created rule? Based on that we can tweak what rules are to be stored in the synchronisation content.

Reply:
Yes I understand, but here it would cause existing functionality to break. Currently we put all rules in sync content including user rules and add them into the cloudRule indexer. If we skip any rules here, update rule routine wouldn't be able to compute and claim the rule, lead to duplicate rule error from AWS, and unnecessary cloud api calls in Azure. We shouldn't skip any rules here, just return all rules and we can add identification of user rules in the next patch

Reply:
Discussed the behavior and this will be addressed in a follow  up patch, as it is not under the scope of this PR.